driver/SDL/sdl_common.c

```diff
@@ -186,25 +182,34 @@ void mousewheel_handler(SDL_Event * event)
  */
 void keyboard_handler(SDL_Event * event)
 {
+    bool key_state = (event->type == SDL_KEYUP) ? LV_INDEV_STATE_RELEASED : LV_INDEV_STATE_PRESSED;
     /* We only care about SDL_KEYDOWN and SDL_TEXTINPUT events */
     switch(event->type) {
         case SDL_KEYDOWN:                       /*Button press*/
+        case SDL_KEYUP:                       /*Button release*/
         {
             const uint32_t ctrl_key = keycode_to_ctrl_key(event->key.keysym.sym);
-            if (ctrl_key == '\0')
-                return;
-            const size_t len = strlen(buf);
-            if (len < KEYBOARD_BUFFER_SIZE - 1) {
-                buf[len] = ctrl_key;
-                buf[len + 1] = '\0';
+            
+            if(key_buffer_len < KEYBOARD_BUFFER_SIZE){
+                key_buffer[key_buffer_len].key = ctrl_key;
+                key_buffer[key_buffer_len].state = key_state;
+                key_buffer_len++;
             }
             break;
         }
         case SDL_TEXTINPUT:                     /*Text input*/
         {
-            const size_t len = strlen(buf) + strlen(event->text.text);
-            if (len < KEYBOARD_BUFFER_SIZE - 1)
-                strcat(buf, event->text.text);
+            const size_t len = key_buffer_len + strlen(event->text.text);
+            if (len < KEYBOARD_BUFFER_SIZE - 1) {
```

Comment:
Now that you are adding both "pressed" and "released" events, I think you also need to change this "if" condition to prevent overflowing the buffer, since you are now adding **two** elements to the buffer for each character and not one.